climpred/checks.py

```diff
@@ -143,3 +156,61 @@ def match_initialized_vars(init, ref):
             f'got {init_vars} for init and {ref_vars} for ref.'
         )
     return True
+
+
+def has_valid_lead_units(xobj):
+
+    if hasattr(xobj['lead'], 'units'):
+
+        if not getattr(xobj['lead'], 'units') in VALID_LEAD_UNITS:
```

Comment:
Just for xtra user friendliness, since all the valid lead units ends with an "s", maybe right before this check:
```
units = getattr(xobj['lead'], 'units')
if not units.endswith('s'):
    units += 's'
    xobj['lead'].attrs['units'] = units
    warnings.warn(f'The letter "s" was appended to the lead units; now {units}.')
```
That way, if units happened to be "year" instead of "years", or 'day' instead of 'days' it could still work?